src/system/AdminModule/Controller/AdminInterfaceController.php

```diff
@@ -0,0 +1,773 @@
+                }
+                if (!array_key_exists('prerelease', $version) || $version['prerelease']) {
+                    continue;
+                }
+                if (array_key_exists('tag_name', $version)) {
+                    if (version_compare($version['tag_name'], $onlineVersion) == 1) {
+                        $onlineVersion = $version['tag_name'];
+                    }
+                }
+            }
+
+            if ($onlineVersion == '') {
+                return array('update_show' => false);
+            }
+            System::setVar('updateversion', $onlineVersion);
+        }
+
+        // if 1 then there is a later version available
+        if (version_compare($onlineVersion, Zikula_Core::VERSION_NUM) == 1) {
+            return array('update_show' => true,
+                         'update_version' => $onlineVersion);
+        } else {
+            return array('update_show' => false);
+        }
+    }
+
+    /**
+     * Developer notices.
+     *
+     * @return array developer notice data or false
+     */
+    private function _developernotices()
+    {
+        $modvars = ModUtil::getVar('ZikulaThemeModule');
+
+        $data = array();
+        $data['devmode'] = $this->get('kernel')->getEnvironment() === 'dev';
+
+        if ($data['devmode'] == true) {
+            $data['cssjscombine'] = $modvars['cssjscombine'];
+
+            if ($modvars['render_compile_check']) {
+                $data['render']['compile_check'] = array('state' => $modvars['render_compile_check'],
+                        'title' => $this->__('Compile check'));
+            }
+            if ($modvars['render_force_compile']) {
+                $data['render']['force_compile'] = array('state' => $modvars['render_force_compile'],
+                        'title' => $this->__('Force compile'));
+            }
+            if ($modvars['render_cache']) {
+                $data['render']['cache']         = array('state' => $modvars['render_cache'],
+                        'title' => $this->__('Caching'));
+            }
+            if ($modvars['compile_check']) {
+                $data['theme']['compile_check']  = array('state' => $modvars['compile_check'],
+                        'title' => $this->__('Compile check'));
+            }
+            if ($modvars['force_compile']) {
+                $data['theme']['force_compile']  = array('state' => $modvars['force_compile'],
+                        'title' => $this->__('Force compile'));
+            }
+            if ($modvars['enablecache']) {
+                $data['theme']['cache']          = array('state' => $modvars['enablecache'],
+                        'title' => $this->__('Caching'));
+            }
+        }
+
+        return $data;
+    }
+
+    /**
+     * Zikula curl
+     *
+     * This function is internal for the time being and may be extended to be a proper library
+     * or find an alternative solution later.
```

Comment:
it might be better to outsource this all to an own class? @cmfcmf did you write this one?